src/main/java/bral/fallingsand/Sand.java

```diff
@@ -0,0 +1,160 @@
+                builder.append(field[y][x]);
+            }
+            builder.append("\n");
+        }
+        return builder.toString();
+    }
+
+    /**
+     * @return the value in field
+     */
+    public int get(int x, int y) {
+        return field[y][x];
+    }
+
+    /**
+     * Sets the value in field to be 1
+     */
+    public void put(int x, int y) {
+        field[y][x] = 1;
+    }
+
+    public void fall() {
+        //moves all sand down one square
+
+        // check every cell
+        boolean moved = true;
+        // while moved is true, needs to continue rechecking all of them
+        while (moved) {
+            moved = false;
+            for (int y = field.length - 2; y >= 0; y--) {
+                for (int x = 0; x < field[y].length; x++) {
+
+                    if (field[y][x] == 1) {
+                        if (field[y + 1][x] == 0) {
+                            // does the sand fall straight down?
+                            field[y][x] = 0;
+                            field[y + 1][x] = 1;
+                            moved = true;
+                            continue;
+                        }
+
+                        boolean rightFirst = random.nextBoolean();
+                        int direction1 = rightFirst ? +1 : -1;
+                        int direction2 = rightFirst ? -1 : +1;
+
+                        if (field[y + 1][x + direction1] == 0) {
+                            field[y][x] = 0;
+                            field[y + 1][x + direction1] = 1;
+                        } else if (field[y + 1][x + direction2] == 0) {
+                            field[y][x] = 0;
+                            field[y + 1][x + direction2] = 1;
+                        }
+                    }
+
+                }
+            }
+        }
+
+    }
+
+    // old code
+                /*if (field[y + 1][x] == 0) {
+                    // does the sand fall straight down?
+                    if (field[y][x] == 1 && field[y + 1][x] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x] = 1;
+                    }
+
+                    // does the sand fall to the right?
+                    if (field[y][x] == 1 && field[y + 1][x + 1] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x + 1] = 1;
+                    }
+
+                    // does the sand fall to the left?
+                    if (field[y][x] == 1 && field[y + 1][x + -1] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x - 1] = 1;
+                    }
+                }*/
+
+    public void randomSand(int n) {
+        //This will add n pieces of sand into the field in random positions
+        /*
+        needs to check where there are empty boxes on top
+        of the empty boxes, choose where to put sand
+        fall
+        repeat
+         */
+        for (int i = 0; i < n; i++) {
```

Comment:
You need to put sand in random position within the whole field, not just one row which is what I think you're doing.

I also never asked you to worry about empty spots.

This is a lot simpler than you are trying to write it.